### ISSUE TITLE
Result of adding noise is very different in mps vs cuda or cpu

### ISSUE THREAD

**Issue by atabakd** [pytorch/pytorch#111169]
### 🐛 Describe the bug

Possibly related to #84936 I was trying to follow [this tutorial](https://colab.research.google.com/drive/1NFxjNI-UIR7Ku0KERmv7Yb_586vHQW43?usp=sharing#scrollTo=F724QP16w7-A), but I noticed the results of adding noise in `cpu/cuda` are quite different than `mps`. This was tested on both `PyTorch 2.2.0.dev20231008` and `PyTorch 2.1.0`.

![cpu_mps](https://github.com/pytorch/pytorch/assets/5020822/da099f7e-d353-4fea-ba47-340c7080c703)

I created a gist to easily replicate the problem here:

https://gist.github.com/atabakd/e4fa20c23f7a6059dac794a82dccfb38

### Versions

```
Collecting environment information...
PyTorch version: 2.1.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 14.0 (arm64)
GCC version: Could not collect
Clang version: 15.0.0 (clang-1500.0.40.1)
CMake version: version 3.27.4
Libc version: N/A

Python version: 3.9.18 | packaged by conda-forge | (main, Aug 30 2023, 03:53:08)  [Clang 15.0.7 ] (64-bit runtime)
Python platform: macOS-14.0-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Apple M1 Max

Versions of relevant libraries:
[pip3] mypy-extensions==1.0.0
[pip3] numpy==1.24.4
[pip3] torch==2.1.0
[pip3] torchaudio==2.1.0
[pip3] torchvision==0.16.0
[conda] numpy                     1.24.4           py39h485cf63_0    conda-forge
[conda] pytorch                   2.1.0                   py3.9_0    pytorch
[conda] torchaudio                2.1.0                  py39_cpu    pytorch
[conda] torchvision               0.16.0                 py39_cpu    pytorch
```

cc @kulinseth @albanD @malfet @DenisVieriu97 @razarmehr @abhudev

**Comment by watarungurunnn** [comment]
According to my investigation, the problem is with the memory_format.

Changing 
`eps = torch.randn_like(xtm1)` to `eps = torch.randn_like(xtm1, memory_format=torch.contiguous_format)`
or
adding `x = x.contiguous(memory_format=torch.contiguous_format)` after `x = img_to_tensor(im)`
, this issue doesn't occur. 

Also, the code below shows that the tensor is allocated in channels_last format.
```python
x = img_to_tensor(im)
print(x.is_contiguous(memory_format=torch.channels_last))
print(x.is_contiguous(memory_format=torch.contiguous_format))
```
```output
True
False
```
Additionally, I show here the output of `torch_randn` in your code, whose min, std should be 0 and 1, respectively.
![fail2_min](https://github.com/pytorch/pytorch/assets/84391804/ba3ccc68-867a-41a4-b05b-9e907bdba077)
![fail2_std](https://github.com/pytorch/pytorch/assets/84391804/fdfb7d47-30dc-4883-aabb-5d02c0e659b7)
![fail2_max](https://github.com/pytorch/pytorch/assets/84391804/b1405a43-f1f8-4a1f-8d0d-a5f94df7ef91)
![fail2_mean](https://github.com/pytorch/pytorch/assets/84391804/9e79bc23-b8da-4f41-9cf1-6f09913e5170)

also, refer to: https://github.com/pytorch/pytorch/issues/62872

**Comment by atabakd** [comment]
I can confirm the issue is resolved in `pytorch 2.2.2` and as announced [here](https://github.com/pytorch/pytorch/issues/111634#issuecomment-2048616049) so I can close the issue.
